src/sql/src/plan.rs

```diff
@@ -932,10 +934,21 @@ pub struct ShowColumnsPlan {
 
 #[derive(Debug)]
 pub struct CopyFromPlan {
+    /// Table we're copying into.
     pub id: CatalogItemId,
+    /// Source we're copying data from.
     pub source: CopyFromSource,
-    pub columns: Vec<usize>,
+    /// How input columns map to those on the destination table.
+    ///
+    /// TODO(cf2): Remove this field in favor of the mfp.
+    pub columns: Vec<ColumnIndex>,
```

Comment:
Sounds like an easy win to fold this into the current PR. Is it difficult to add the projection into the MFP?

Reply:
It would be a bit tricky to fold this into the current PR because the current `COPY FROM ... STDIN` still uses these columns. We have https://github.com/MaterializeInc/database-issues/issues/8859 tracking the work to fold all of that into this new setup